src/Codeception/Coverage/Subscriber/LocalServer.php

```diff
@@ -148,18 +149,52 @@ function ($h) {
 
     protected function startCoverageCollection($testName)
     {
-        $cookie = [
+        $value = [
             'CodeCoverage'        => $testName,
             'CodeCoverage_Suite'  => $this->suiteName,
             'CodeCoverage_Config' => $this->settings['remote_config']
         ];
+        $value = json_encode($value);
+
         $this->module->amOnPage('/');
-        $this->module->setCookie(self::COVERAGE_COOKIE, json_encode($cookie));
+        $this->module->setCookie(self::COVERAGE_COOKIE, $value);
+
+        // putting in configuration ensures the cookie is used for all sessions of a MultiSession test
+
+        $cookies = $this->module->_getConfig('cookies');
+        if (!$cookies|| !is_array($cookies)) {
+            $cookies = [];
+        }
+
+        $cookie = null;
+        foreach ($cookies as &$cookie) {
+            if (!is_array($cookie) || !array_key_exists('Name', $cookie) || !array_key_exists('Value', $cookie)) {
+                // \Codeception\Lib\InnerBrowser will complain about this
+                continue;
+            }
+            if ($cookie['Name'] === self::COVERAGE_COOKIE) {
+                break;
+            }
+        }
+        if (!$cookie || $cookie['Name'] !== self::COVERAGE_COOKIE) {
+            $cookie = &$cookies[];
+            $cookie['Name'] = self::COVERAGE_COOKIE;
+        }
+
+        $cookie['Value'] = $value;
+
+        $this->module->_setConfig(['cookies' => $cookies]);
```

Comment:
I agree with this way but could the code above be simplified or documented? 
This things especially with `&` are pretty hard to understand.